src/HealthGPS/energy_balance_model.cpp

```diff
@@ -81,6 +88,18 @@ void EnergyBalanceModel::update_risk_factors(RuntimeContext &context) {
     }
 }
 
+double EnergyBalanceModel::bounded_nutrient_value(const core::Identifier &nutrient,
+                                                  double value) const {
+    auto &range = nutrient_ranges_.at(nutrient);
+    if (value < range.first) {
+        return range.first;
+    }
+    if (value > range.second) {
+        return range.second;
+    }
+    return value;
```

Comment:
Are you sure you want to cap the values rather than throwing an error if the values are out of bounds?

If so, you can write this as so:

```suggestion
    const auto &range = nutrient_ranges_.at(nutrient)
    return std::clamp(range.first, range.second, value);
```

You need to `#include <algorithm>` for this.

Reply:
This is an interesting question, and one that should be bought up next meeting. The existing code merely caps without error. Maybe there's some argument of homeostasis or digestion capacity. I'll bring it up.

Reply:
Maybe open a GitHub issue for it too so we don't forget about it?

From the bits of the code I've looked at, it seems that doing this sort of thing (i.e. carrying on in the face of unexpected input) is fairly common, but I don't think it's good. It'd be better to give the user a big fat error message and then stop IMO, or at least give them a warning!